packages/trace-viewer/src/ui/uiModeTraceView.tsx

```diff
@@ -73,9 +73,31 @@ export const TraceView: React.FC<{
     pollTimer.current = setTimeout(async () => {
       try {
         const model = await loadSingleTraceFile(traceLocation);
-        setModel({ model, isLive: true });
-      } catch {
-        setModel(undefined);
+        setTrace({ type: 'model', model, isLive: true });
+      } catch (e) {
+        if (result.status !== 'failed') {
+          setTrace(undefined);
+          return;
+        }
+
+        // Test failed. We don't know what happened to the trace
+        const treeErrors = () => {
+          if (!item.treeItem || item.treeItem.kind !== 'case' || !item.treeItem.test)
+            return [];
+          const test = item.treeItem.test;
```

Comment:
- Why are we looking at all results?
- Why not compute as a memo, why callback?